workspaces/ballerina/ballerina-extension/src/features/project/cmds/cloud.ts

```diff
@@ -25,61 +25,85 @@ import {
     commands.registerCommand(PALETTE_COMMANDS.CLOUD, async () => {
         try {
             sendTelemetryEvent(extension.ballerinaExtInstance, TM_EVENT_PROJECT_CLOUD, CMP_PROJECT_CLOUD);
 
-            if (window.activeTextEditor && window.activeTextEditor.document.languageId != LANGUAGE.BALLERINA) {
+            if (window.activeTextEditor && window.activeTextEditor.document.languageId !== LANGUAGE.BALLERINA) {
                 window.showErrorMessage(MESSAGES.NOT_IN_PROJECT);
                 return;
             }
+ 
+            const result = await findWorkspaceTypeFromWorkspaceFolders();
+            let { projectPath, projectInfo } = StateMachine.context();
 
-            const isDiagram: boolean = extension.ballerinaExtInstance.getDocumentContext().isActiveDiagram();
-           
-            const currentProject = isDiagram ? await
-                getCurrentBallerinaProject(extension.ballerinaExtInstance.getDocumentContext().getLatestDocument()?.toString())
-                : await getCurrentBallerinaProject();
-
-            if (currentProject.kind !== PROJECT_TYPE.SINGLE_FILE) {
-                if (currentProject.path) {
-                    let cloudTomlPath = currentProject.path + CLOUD_CONFIG_FILE_NAME;
-                    if (!fs.existsSync(cloudTomlPath)) {
-                        const commandArgs = {
-                            key: "uri",
-                            value: isDiagram ? extension.ballerinaExtInstance.getDocumentContext().getLatestDocument()?.toString()
-                                : window.activeTextEditor!.document.uri.toString()
-                        };
-                        commands.executeCommand('ballerina.create.cloud.exec', commandArgs);
-                        outputChannel.appendLine(`Cloud.toml created in ${currentProject.path}`);
-                    } else {
-                        const message = `Cloud.toml already exists in the project.`;
-                        sendTelemetryEvent(extension.ballerinaExtInstance, TM_EVENT_ERROR_EXECUTE_PROJECT_CLOUD,
-                            CMP_PROJECT_CLOUD, getMessageObject(message));
-                        window.showErrorMessage(message);
+            let targetPath = projectPath ?? "";
+            if (result.type !== "SINGLE_PROJECT") {
+                if (result.type === "MULTIPLE_PROJECTS") {
+                    const packageRoot = await getCurrentProjectRoot();
+                    if (!packageRoot) {
+                        window.showErrorMessage(MESSAGES.NO_PROJECT_FOUND);
+                        return;
+                    }
+                    projectInfo = await StateMachine.langClient().getProjectInfo({ projectPath: packageRoot });
+                    targetPath = projectInfo.projectPath ?? packageRoot;
+                } else if (result.type === "BALLERINA_WORKSPACE") {
+                    const selection = await getPackage(projectInfo, "Select the project to create Cloud.toml in");
```

Comment:
With this logic, the project picker will appear even when the user is focused on a specific package within the workspace. Ideally, the picker should be shown only in the following cases:

1. When the command is executed from the workspace overview.
2. When a workspace is open but no webview or text editor is active.

We can follow the approach used in [this implementation](https://github.com/vinukab/vscode-extensions/blob/91e7323d988448a34654aa229585f94300f33aa3/workspaces/ballerina/ballerina-extension/src/features/project/cmds/doc.ts#L69-L84) to handle these scenarios more accurately.